relations/src/gr1cs/constraint_system.rs

```diff
@@ -275,14 +284,26 @@ impl<F: Field> ConstraintSystem<F> {
             let assignments = &mut self.assignments;
 
             let lc_indices = lcs.into_iter().map(|lc| {
-                let index = LcIndex(*num_lcs);
-                lc_map.push(Some(lc));
-                *num_lcs += 1;
-                if should_generate_lc_assignments {
-                    let value = assignments.eval_lc(index, lc_map).unwrap();
-                    assignments.lc_assignment.push(value);
+                let lc = lc();
+                match lc.0.as_slice() {
+                    // If the linear combination is empty, we return a symbolic LC with index 0.
+                    // If the linear combination is just Zero, we return a symbolic LC with index 0.
+                    [] | [(_, Variable::Zero)] => Variable::SymbolicLc(LcIndex(0)),
```

Comment:
If the lc is empty or consists of a single element with `Variable::Zero`, then we return the index for the zero LC.